replication/row_event.go

```diff
@@ -1141,17 +1145,34 @@ func (e *RowsEvent) decodeValue(data []byte, tp byte, meta uint16) (v interface{
 		v, err = littleDecodeBit(data, nbits, n)
 	case MYSQL_TYPE_BLOB:
 		v, n, err = decodeBlob(data, meta)
+		newValue, ok := convertToString(v)
+		if ok {
+			v = newValue
+		}
 	case MYSQL_TYPE_VARCHAR,
 		MYSQL_TYPE_VAR_STRING:
 		length = int(meta)
-		v, n = decodeString(data, length)
+		fmt.Printf("islatin is: %v", e.Table.isLatin)
+		if !utf8.Valid(data) && e.Table.isLatin {
+			v, n = decodeStringLatin1(data, length)
+		} else {
+			v, n = decodeString(data, length)
+		}
 	case MYSQL_TYPE_STRING:
-		v, n = decodeString(data, length)
+		if !utf8.Valid(data) && e.Table.isLatin {
+			v, n = decodeStringLatin1(data, length)
```

Comment:
The decoding and converting should be done at upper layer, not here. `replication` package aims to provide a golang representation of binlog.